src/main/scala/cromwell/util/google/GoogleCredentialFactory.scala

```diff
@@ -114,4 +116,14 @@ abstract class GoogleCredentialFactory {
       case unrecognized => Failure(new IllegalArgumentException(s"Unrecognized userSchemeAuthentication: $unrecognized"))
     } getOrElse Failure(new Throwable("No user authentication configuration has been found."))
   }
+
+  private def forApplicationDefaultCredentials(): Credential = {
+    try {
+      GoogleCredential.getApplicationDefault().createScoped(GoogleScopes.Scopes.asJava)
+    } catch {
+      case e : IOException =>
+      log.warn("Failed to get application default credentials", e)
+      throw e
```

Comment:
Thanks. What about that failing integration test?

On Wed, Jan 6, 2016 at 2:56 PM, mcovarr <notifications@github.com> wrote:

> In src/main/scala/cromwell/util/google/GoogleCredentialFactory.scala
> <https://github.com/broadinstitute/cromwell/pull/356#discussion_r49001789>
> :
>
> > @@ -114,4 +116,14 @@ abstract class GoogleCredentialFactory {
> >        case unrecognized => Failure(new IllegalArgumentException(s"Unrecognized userSchemeAuthentication: $unrecognized"))
> >      } getOrElse Failure(new Throwable("No user authentication configuration has been found."))
> >    }
> > +
> > +  private def forApplicationDefaultCredentials(): Credential = {
> > +    try {
> > +      GoogleCredential.getApplicationDefault().createScoped(GoogleScopes.Scopes.asJava)
> > +    } catch {
> > +      case e : IOException =>
> > +      log.warn("Failed to get application default credentials", e)
> > +      throw e
>
> OK, it's possible the caller wasn't logging the exception. Unfortunately I
> don't think we have a policy on how this ought to be done in Cromwell.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/cromwell/pull/356/files#r49001789>.
>